index.html

```diff
@@ -7714,6 +7714,80 @@ <h3>Processing actions</h3>
  input JSON, such that the actions to be performed in a single <a>tick</a>
  are grouped together.
 
+<p>An <dfn>element origin</dfn> is a Map
+which [=map/exist|contains=] "<code>type</code>" and
+"<code>sharedId</code>", and for which the value of the entry with key
+"<code>type</code>" is "<code>node</code>".
+
+<p>To <dfn>get coordinates relative to an origin</dfn>
+given <var>source</var>, <var>x offset</var>, <var>y offset</var>,
+and <var>origin</var>:
+
+<ol class="algorithm">
+ <li><p>Run the substeps of the first matching value
+  of <var>origin</var>
+  <p>Assert: one of these conditions matches
+
+  <dl>
+   <dt>"<code>viewport</code>"
+   <dd>
+    <ol>
+     <li><p>Let <var>x</var> equal <var>x offset</var> and
+     <var>y</var> equal <var>y offset</var>.
+    </ol>
+   </dd>
+
+   <dt>"<code>pointer</code>"
+   <dd>
+    <p class="note">This is only possible if <var>source</var> is
+    a <a>pointer input source</a></p>
+
+    <ol>
+     <li><p>Let <var>start x</var> be equal to the <code>x</code>
+     property of <var>source</var>.
+
+     <li><p>Let <var>start y</var> be equal to the <code>y</code>
+     property of <var>source</var>.
+
+     <li><p>Let <var>x</var> equal <var>start x</var> +
+     <var>x offset</var> and <var>y</var> equal
+     <var>start y</var> + <var>y  offset</var>.
+    </ol>
+   </dd>
+   <dt>An <a>element origin</a>, or an object that <a>represents a web element</a></dt>
```

Comment:
I'm a bit unclear about what exactly this check is. Following the definition of represents a web element, would this match for the object `{ "element-6066-11e4-a52e-4f735466cecf": "whatever" }`?

It would be good to spell out what the element origin type test looks like, in particular if we tolerate additional keys in the map beyond "type" and "sharedId".